torch_geometric/nn/aggr/basic.py

```diff
@@ -66,6 +66,130 @@ def forward(self, x: Tensor, index: Optional[Tensor] = None,
         return torch.sqrt(var.relu() + 1e-5)
 
 
+class QuantileAggregation(Aggregation):
+    r"""Quantile aggregation operator.
+
+    Args:
+        q (float): A scalar in the range [0, 1].
+        interpolation (str): Interpolation method applied if the quantile point
+            :math:`q(n - 1)` lies between two values :math:`x_i \le x_{i+1}`,
+            with :math:`i < q(n - 1) < i+1`. Can be one of the following:
+
+            - :obj:`"lower"`: Returns the one with lowest value.
+
+            - :obj:`"higher"`: Returns the one with highest value.
+
+            - :obj:`"midpoint"`: Returns the average of the two values.
+
+            - :obj:`"nearest"`: Returns the one nearest to the quantile point.
+
+            - :obj:`"linear"` (default): Returns a linear combination of the
+                two elments, as
+                :math:`x_i + (x_{i+1} - x_i)\cdot(q(n - 1) - i)`.
+
+        fill_value (float): Default value in the case no entry is
+            found for a given index (default: :obj:`NaN`).
+    """
+
+    interpolations = {'linear', 'lower', 'higher', 'nearest', 'midpoint'}
+
+    def __init__(self, q: float, interpolation: str = 'linear',
+                 fill_value: float = float('nan')):
+        if q > 1. or q < 0:
+            raise ValueError("'q' must be in the range [0, 1].")
+        if interpolation not in self.interpolations:
+            raise ValueError(f"Invalid '{interpolation}' interpolation method."
+                             f" Available interpolations:"
+                             f"{self.interpolations}")
+
+        super().__init__()
+        self.q = q
+        self.interpolation = interpolation
+        self.fill_value = fill_value
+
+    def forward(self, x: Tensor, index: Optional[Tensor] = None,
+                ptr: Optional[Tensor] = None, dim_size: Optional[int] = None,
+                dim: int = -2) -> Tensor:
+        # we could also expand `ptr` to an index
```

Comment:
We only support `index` for many of the aggregations already,  seems fine to me.

Reply:
Ok, I'll remove the comment!